### PR TITLE
properly convert/send image type to extension

### PR DESCRIPTION
**Pull request by justschen**
re: plumbing for image data in the extension

in the extension looks like this:
Screenshot 2024-09-30 at 2 25 06 PM

### CODE REVIEW

File: src/vscode-dts/vscode.d.ts [hunk 1]
```diff
@@ -19046,7 +19046,29 @@ declare module 'vscode' {
...
 		 * The value of this reference. The `string | Uri | Location` types are used today, but this could expand in the future.
 		 */
-		readonly value: string | Uri | Location | unknown;
+		readonly value: string | Uri | Location | ChatReferenceBinaryData | unknown;
+	}
+
+	/**
+	 * Represents binary data associated with a chat reference.
+	 */
+	export class ChatReferenceBinaryData {
```
Comment (mjbvz) [inline src/vscode-dts/vscode.d.ts:19055]: Worth thinking about how we can defer actually sending the data until it is needed. See how `DataTransferFile` for example. We also allow `DataTransferFile` to be passed to a workspace edit which means we never need to transfer the data to extensions

File: src/vscode-dts/vscode.d.ts [hunk 2]
```diff
@@ -19046,7 +19046,29 @@ declare module 'vscode' {
...
+		* The binary data of the reference.
+		*/
+		readonly data: Uint8Array;
+
+		/**
+		 * Creates a new instance of ChatReferenceBinaryData.
+		 * @param mimeType The MIME type of the binary data.
+		 * @param data The binary data of the reference.
+		 */
+		constructor(mimeType: string, data: Uint8Array);
```
Comment (mjbvz) [inline src/vscode-dts/vscode.d.ts:19071]: Please make sure this does not get merged until it goes through api review too. Is it possible to ship as a proposal instead?

File: src/vs/workbench/api/common/extHostTypeConverters.ts [hunk 3]
```diff
@@ -2796,7 +2796,7 @@ export namespace ChatPromptReference {
 			range: variable.range && [variable.range.start, variable.range.endExclusive],
 			value: isUriComponents(value) ? URI.revive(value) :
 				value && typeof value === 'object' && 'uri' in value && 'range' in value && isUriComponents(value.uri) ?
-					Location.to(revive(value)) : value,
+					Location.to(revive(value)) : variable.isImage ? new types.ChatReferenceBinaryData('image/*', new Uint8Array(Object.values(value))) : value,
```
Comment (mjbvz) [inline src/vs/workbench/api/common/extHostTypeConverters.ts:2799]: Make sure to get the proper mime here instead of the generic `image/*` one

Comment (justschen) [reply src/vs/workbench/api/common/extHostTypeConverters.ts]: yessir, waiting for https://github.com/microsoft/vscode/pull/229779 which let's us get the correct type (previously, it was almost always converted into `image/png`)

**Review by rzhao271 — APPROVED** [review]

**Pull request merged**